### ISSUE TITLE
Underscores in identity prevent any tracking.

### ISSUE THREAD

**Issue by mmdock** [RevenueCat/purchases-ios#249]
Recently setup observer mode for RevenueCat.  Encountered a strange issue: when setting identity of the user, any identity with an underscore character was not being recorded.  I was unable to find any documentation regarding this or other special characters to be wary about. Luckily, we have yet to release this.

Is there any documentation regarding unusable characters for setting identity? The documentation  for identifying users here: https://docs.revenuecat.com/docs/user-ids mentions nothing specific to watch out for, and in fact, the fake id "my_app_user_id" has underscores, making this more funny.

In Summary: 
- Tried purchasing with ids that have no underscore.  Tracking worked
- Tried purchasing with ids that have an underscore.  Tracking did not work.  Even waiting a day for confirmation just in case. No purchase was recorded, no new users, nothing.  
- documentation about special characters need.

Thanks!

**Comment by michaellenaghan** [comment]
Thanks for reporting this; I'm using CloudKit user record identifiers for identity, and they start with an underscore!

**Comment by vegaro** [comment]
I have just successfully purchased using a app user id `_user_with_underscore` and I am seeing the purchases in the RevenueCat dashboard. Maybe I am misunderstanding the issue here.

**Comment by alfonsoperez** [comment]
Hey @mmdock, @michaellenaghan @cuity it seems that this issue is not widespread to other users, so could you please file a bug report in here: https://support.revenuecat.com/hc/en-us/articles/360046346873-Bug-Reports, sharing if possible a reproducible example so we can take a deeper look at your case in particular?.

I am interested mainly in a [Debug log](https://docs.revenuecat.com/docs/debugging) and a network log to see what request is actually doing. Otherwise if you try to reproduce again, let us know in the ticket your details and the app user id of the example so we can check on our server logs.

Thank you!

**Comment by michaellenaghan** [comment]
Sorry for the confusion, but I wasn't reporting a problem, I was just thanking mmdock for reporting it—and indirectly pointing out that anyone using CloudKit user record identifiers would have the same problem. (I was thinking of looking into this myself next week; I haven't completed my own RC integration yet.)

**Comment by aboedo** [comment]
@mmdock have you managed to figure this out?

**Comment by aboedo** [comment]
I'm closing this ticket due to inactivity. Feel free to reopen if something comes up related to this, though.
